modin/modin_logging/config.py

```diff
@@ -0,0 +1,143 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import logging
+import datetime as dt
+import os
+import uuid
+import platform
+import psutil
+import pkg_resources
+import threading
+import time
+from modin.config import LogMemoryInterval, LogMode
+
+__LOGGER_CONFIGURED__: bool = False
+
+
+class ModinFormatter(logging.Formatter):
+    def formatTime(self, record, datefmt=None):
```

Comment:
From the docs for the Formatter class:
> formatTime(record, datefmt=None)
> This method should be called from [format()](https://docs.python.org/3/library/functions.html#format) by a formatter which wants to make use of a formatted time. This method can be overridden in formatters to provide for any specific requirement, but the basic behavior is as follows: if datefmt (a string) is specified, it is used with [time.strftime()](https://docs.python.org/3/library/time.html#time.strftime) to format the creation time of the record. Otherwise, the format ‘%Y-%m-%d %H:%M:%S,uuu’ is used, where the uuu part is a millisecond value and the other letters are as per the time.strftime() documentation. An example time in this format is 2003-01-23 00:29:50,411. The resulting string is returned.

Why are we redefining this method if it does the same thing as its parent method?

Reply:
The parent method is returning a millisecond value, this is being overridden to record microsecond timestamps.